basket/news/tasks.py

```diff
@@ -928,6 +928,12 @@ def process_donation(data):
             raise
 
 
+@et_task
+def process_newsletter_subscribe(data):
+    data['lang'] = get_best_supported_lang(data['lang'])
```

Comment:
Is there any need to ensure `lang` exists in `data`? Or that `data` contains all info needed to `upsert_user`?

Reply:
It should all always be there. See the code on the mofo side:

https://github.com/mozilla/foundation.mozilla.org/blob/060922af4dd07f35f8119f2596228f47946b4487/network-api/networkapi/campaign/views.py#L87-L93

I really only did the `best_supported_lang` call to prepare for them someday sending a real language. It's hard coded to `en` for now.